app/models/piece.rb

```diff
@@ -1,3 +1,130 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  # anywhere it say y_coord was self.y & x_coord was self.x
+
+                # Need function to automoatically keep up with current pieces coords at all times.
+  def current_pieces_coords    # Will need to be removed later after new function made.  
+    [[0,7],[3,7],[4,7],[5,7],[6,7],[7,7],
+    [2,6],[3,6],[4,6],[5,6],[6,6],[7,6],
+    [0,5],[1,5],[7,5],
+    [2,3],[3,3],
+    [0,1],[1,1],[4,1],[5,1],[6,1],[7,1],
+    [0,0],[1,0],[3,0],[5,0],[6,0],[7,0]]      # Represents all the coordinates of pieces in play
+  end
+
+  def is_obstructed?(x, y)
+    if x_coord == x
```

Comment:
what if x_coord == x AND y_coord == y?

Reply:
What if we say if `x_coord == x && y_coord != y`? Then we know for sure that we are dealing with a vertical obstruction? same for horiz/diag...

Reply:
Exactly!

Reply:
This works perfect! Thanks @bhk1019! & @ilyakrasnov.